### PR TITLE
docs: consolidate roadmaps into docs/roadmap/ + CHANGELOG

### PR DESCRIPTION
**Pull request by robcohen**
## Consolidate the roadmaps into one forward-only system

Reorganizes the four scattered, drift-prone roadmaps into a single structure. **Includes the accuracy fixes from #1372** (which this supersedes) — the corrected content was carried into the new files.

### The structural fix
The roadmaps kept drifting because **status lived inline** (done items piled up as stale "✅" rows). The fix: roadmaps are now **forward-only**, and shipped work moves to a **`CHANGELOG.md`**.

### New layout
```
docs/roadmap/
  index.md                  # entry point: themes, lifecycle, tracked work
  performance.md            # forward-only (Now / Next / Exploring)
  importing.md
  testing-and-quality.md
  formal-verification.md    # absorbs spec/tla/ROADMAP.md
CHANGELOG.md                # shipped work, by area
docs/development/import-architecture.md   # deep import design, split out
```
- Old roadmap paths → thin **redirect stubs**; `docs/development/index.md` repointed.
- CHANGELOG links use the GitHub URL (it's repo-root, not synced to the docs site).
- **Brainstormed/deferred items folded in** under "Exploring": MCP server, browser playground, incremental LSP reparse, shareable importer registry, plus the deferred engineering work (dylint + unwrap/panic cleanup, loom, promote SemVer gate + Tier-2 + cargo-public-api, plugin commutativity).

### Companion
`rustledger/rustledger.github.io#14` adds a **Roadmap** section to the site sidebar (the roadmaps were never in the published nav before — this surfaces them).

Closes #1372 (superseded).

### CODE REVIEW

File: docs/roadmap/performance.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Forward-looking performance work. rustledger is already 10-30x faster than Python beancount on typical ledgers; this tracks the remaining optimizations and ideas, not what has already shipped.
+
+## Now / In progress
+
+Items that are partially done or the most valuable next steps.
+
+| Item | Notes |
+|------|-------|
+| Bumpalo arena for AST nodes | Phase 6 (lexer + arena) is partial: the Logos lexer and structured CST parser shipped, but AST allocation still uses the global allocator. Move AST nodes into a [bumpalo](https://github.com/fitzgen/bumpalo) arena (~11 instructions/alloc, mass-reset on discard) for phase-oriented parse → use → discard. Projected ~+20% parsing. |
```
[the remark this answers is not in the document]
Comment (robcohen) [reply docs/roadmap/performance.md:13]: Fixed — "Projected ~+20% parsing" -> "Projected ~20% faster parsing".

File: docs/roadmap/testing-and-quality.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Now / In progress
+
+Partially-done or actively-worth-doing-next work that builds on infrastructure
+already in place.
+
+| Item | Notes |
+|------|-------|
+| Promote `semver-plugin-types` to a **required** check | The `cargo-semver-checks` gate on `rustledger-plugin-types` runs on every PR (#1233); make it a required status check so accidental breaking changes to the published plugin DTOs can't be merged. |
+| **unwrap/panic-in-lib cleanup** | Sweep library crates for `.unwrap()`/`panic!` in non-test code and replace with `Result`/`?` (per the project's "no `.unwrap()` in library code" rule). Deferred from #1237. |
```
[the remark this answers is not in the document]
Comment (robcohen) [reply docs/roadmap/testing-and-quality.md:16]: Reworded — framed as the CLAUDE.md self-review convention (not an enforced rule); the item is precisely about making it a structural check.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- **Phase 6 — WASM testing:** `wasm-pack` Node.js + browser tests for `rustledger-wasm` (`wasm.yml`, `tests/node.rs`).
+- **Pipeline-boundary property tests (#1235):** `pipeline_invariants.rs` across parser/booking/validate/query/plugin, plus `booking_phase_invariants.rs`, `tla_proptest.rs`, and `plugin_determinism.rs` — pinning parse/format roundtrip + idempotence, booking idempotence, validation determinism, plugin wire-format roundtrip, and query-result determinism.
+- **Grep ratchets (#1237):** `check-sync-primitives.sh` (forbids `std::sync::Mutex`/`RwLock` in library code) and `check-hot-path-collections.sh` (forbids SipHash `HashMap`/`HashSet` in `fxhash-only` modules), wired into `ci.yml` alongside the existing `check-unsafe-invariant.sh` gate.
+- **SemVer / API-stability gate (#1233):** `semver-plugin-types` CI job runs `cargo-semver-checks` against `rustledger-plugin-types` (Tier-1 plugin DTOs) on every PR.
+
+### Formal Verification
+
+- Built 18 TLA+ specifications, 16 of which are model-checked on every CI run (`.github/workflows/tla.yml`).
```
[the remark this answers is not in the document]
Comment (robcohen) [reply CHANGELOG.md:54]: Fixed — reworded to "16 TLA+ specifications model-checked on every CI run (19 specs total, including demo and historical specs)".

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- WASM import plugins: third-party importers as sandboxed `.wasm` modules
+- Balance-directive generation via `rledger extract --balance`/`--balance-date` (amount user-supplied; automatic statement extraction still future)
+
+### Testing & Engineering Quality
+
+- **Phase 1 — Quick Wins:** Miri weekly UB detection (`miri.yml`), `cargo-nextest` for fast parallel CI tests, Codecov coverage reporting (`quality.yml`), and Criterion micro-benchmarks in CI (`bench-pr.yml`).
+- **Phase 2 — Fuzzing infrastructure:** CI fuzzing (`fuzz.yml`) with `fuzz_parse`, `fuzz_parse_line`, `fuzz_query_parse` (query engine), and `fuzz_booking` targets, running on parser PRs and nightly.
+- **Phase 3 — Compatibility enhancements:** Expanded BQL compat corpus to ~17 curated queries (`bql-queries.toml`) covering aggregates, date/string functions, and edge cases; file sampling via `compat-bql-test.py` (`MAX_FILES = 30`, plugin fixtures prioritized).
+- **Phase 4 — Formal verification bridge:** Kani proof harnesses (`crates/rustledger-core/src/kani_proofs.rs`, `kani.yml`) verifying core invariants, plus TLA+ trace-to-test automation wired into `tla.yml` (19 specs).
```
[the remark this answers is not in the document]
Comment (robcohen) [reply CHANGELOG.md:45]: Fixed — TLA spec counts are now consistent: 16 model-checked in CI of 19 total.
